net: openthread: implement `otPlatRadioSetRxOnWhenIdle`

edmont opened the pull request:
OpenThread upmerge to commit `193e77e`.

Implement `otPlatRadioSetRxOnWhenIdle`  and add `IEEE802154_RX_ON_WHEN_IDLE` capability.

Depends on:

- [x] https://github.com/openthread/openthread/pull/9554

edmont commented:
I'll fix compliance check when updating manifest SHA.

rlubos commented on west.yml:
```diff
@@ -301,7 +301,7 @@ manifest:
       path: modules/lib/open-amp
     - name: openthread
-      revision: 6edb06e4e0472411200ce2a084a783eaf3faffe3
+      revision: pull/122/head
```
`193e77e40ec2387d458eaebd1e03902d86f484a5`

rlubos reviewed (commented):
Can you please split `ieee802154_radio.h` and OT radio platform changes into separate commits? Thanks

edmont committed:
net: openthread: implement `otPlatRadioSetRxOnWhenIdle`

OpenThread upmerge to commit `193e77e`.

Implement `otPlatRadioSetRxOnWhenIdle` radio platform API.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

edmont committed:
drivers: ieee802154: add `IEEE802154_RX_ON_WHEN_IDLE` capability

Introduce `IEEE802154_RX_ON_WHEN_IDLE` capability.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

edmont commented:
Fixed remaining typo in docs.
